models/discordactions.js

```diff
@@ -114,10 +114,41 @@ const updateDiscordImageForVerification = async (userDiscordId) => {
   }
 };
 
+const getNumberOfMemberForGroups = async (groups = []) => {
+  try {
+    if (!groups.length) {
+      return [];
+    }
+    const roleIds = groups.map((group) => group.roleid);
+
+    const snapshots = await memberRoleModel.where("roleid", "in", roleIds).get();
+    const roleCount = {};
+
+    snapshots.forEach((doc) => {
+      const roleToMemberMapping = doc.data();
+
+      if (roleCount[roleToMemberMapping.roleid]) {
+        roleCount[roleToMemberMapping.roleid] += 1;
+      } else {
+        roleCount[roleToMemberMapping.roleid] = 1;
+      }
+    });
```

Comment:
why do we need this the query snapshot returned by Firestore is already size, we can just query based on single role id and create a map

Reply:
Can you explain in more detail? I am not able to understand what you are trying to say here.

Reply:
I am following the below documentation for where with "in" condition. 
https://firebase.google.com/docs/firestore/query-data/queries#in_not-in_and_array-contains-any

And then going through each array to check which ids are present and how many times.

Reply:
Resolving this conversation in favor of the above comments. 
Please reopen if it feels there is something we can improve on this.

Thanks for reviewing.